docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,172 @@
+Centralized CLI
+===============
+
+Context
+-------
+
+MetalK8s comes with a set of services to operate and monitor the K8s cluster.
+All operations that need to be performed by the Platform Administrator could be
+categorized as follow:
+
+- Cluster Resources Administration (Nodes, Volumes, Deployments, ...)
+- Cluster Administration (Install, Upgrade, Downgrade, Backup, Restore, ...)
+- Solution Administration (CRUD Environment, Import/Remove Solution, ...)
+- Cluster Service Administration (Configure Dex, Prometheus, Alert Manager,
+  ...)
+
+K8s provides the kubectl CLI, which is distributed and enables
+all kind of interaction with all Kubernetes resources, through apiserver, but
+its usage often requires to build verbose YAML files.
+
+Currently, MetalK8s provides other set of scripts or manual procedures, but
+those are located in various locations, their usage may vary and they are not
+developed using the same logic.
+
+This makes the CLI and associated documentation not super intuitive and it
+also makes the maintenance more expensive in the long term.
+
+The goal of the project is to provide MetalK8s administrator with an intuitive
+and easy to use set of tools in order to administrate and operate a finite set
+of functionalities.
+
+Because kubectl is already in place and is well known by Kubernetes
+administrators, it will be used as a standard to follow for all other MetalK8s
+CLIs:
+
+- CLI follows kubectl style: kubectl <action> <resource>
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI provides <action> help when the command is not valid
+- CLI is not interactive (except if password input is needed)
+- CLI should not require password input
+- CLI provides a dryrun mode for intrusive operations
+- CLI provides a verbose (or debug) mode
+- CLI implementation relies on secure APIs
+- CLI support <action> completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+- CLI can be executed from outside of the cluster (except maybe for very first
+  install script)
+
+When it is possible, it would make sense to leverage kubectl plugin
+
+All functionalities are exposed through 2 distincts CLI:
+- kubectl (enriched using kubectl approach)
+- mk8sctl: a new CLI, exposing specific MetalK8s functionalities
+
+In order to operate the cluster with mk8sctl from outside of the cluster, a
```

Comment:
- For any `kubectl` extra functionality, some binary/binaries need to be installed anyway as well.
- I'd say: `metalk8scli` for on-node actions, `kubectl` plugin for anything that can be done remotely.